spec/system/errors_spec.rb

```diff
@@ -1,32 +1,161 @@
-  it 'shows the generic error page' do
-    allow(ApplicationSearchFilter).to receive(:new) {
-      raise StandardError
-    }
-
-    visit '/'
-    click_on 'Search'
-    expect(page).to have_content 'Sorry, something went wrong with our service'
-    expect(page).to have_css('nav.moj-primary-navigation')
-  end
+  context 'when authenticated' do
+    context 'when an unhandled error occurs' do
+      before do
+        allow(ApplicationSearchFilter).to receive(:new) {
+          raise StandardError
+        }
+
+        visit '/'
+        click_on 'Search'
+      end
+
+      it 'shows the unhandled error page and status' do
+        expect(page).to have_content 'Sorry, something went wrong with our service'
+        expect(page).to have_http_status(:internal_server_error)
+      end
+
+      it 'uses the simplified error page layout' do
+        expect(page).not_to have_css('nav.moj-primary-navigation')
+      end
+    end
+
+    context 'when crime application is not found' do
+      before do
+        visit '/applications/123'
+      end
+
+      it 'shows the application not found error page' do
+        expect(page).to have_content "If you're looking for a specific application, go to all open applications."
+        expect(page).to have_http_status(:not_found)
+      end
+
+      it 'uses the system user layout with navigation' do
+        expect(page).to have_css('nav.moj-primary-navigation')
+      end
+    end
+
+    context 'when crime application is found' do
+      include_context 'with an existing application'
+
+      before do
+        visit "/applications/#{crime_application_id}"
+      end
+
+      it 'shows the application' do
+        expect(page).to have_content 'AJ 12 34 56'
+      end
+    end
+
+    context 'when visiting a non existent page' do
+      before do
+        visit '/not/a/page'
+      end
+
+      it 'shows not found error page' do
+        expect(page).to have_content 'If the web address is correct or you selected a link or button'
+        expect(page).to have_http_status(:not_found)
+      end
+
+      it 'uses the simplified errors page layout' do
+        expect(page).not_to have_css('nav.moj-primary-navigation')
+      end
+    end
 
-  it 'shows application not found error page' do
-    visit '/applications/123'
-    expect(page).to have_content "If you're looking for a specific application, go to all open applications."
-    expect(page).to have_css('nav.moj-primary-navigation')
+    context 'when visiting a forbidden page' do
+      before do
+        visit users_auth_failure_path
+      end
+
+      it 'shows the forbidden page' do
+        expect(page).to have_content 'Access to this service is restricted'
+        expect(page).to have_http_status(:forbidden)
+      end
+
+      it 'uses the simplified errors page layout' do
+        expect(page).not_to have_css('nav.moj-primary-navigation')
+      end
+    end
+
+    context 'when visiting an admin page to manage a user' do
```

Comment:
Might be good to make this more explicit - "when visiting an admin page to manage a user as a non-manager user"

Reply:
This is wrapped in the context "when not authenticated"